### PR TITLE
Add post-install scripts to all tests to capture failure details in GitHub workflow summary

### PR DESCRIPTION
**Pull request by marcofranssen**
See https://github.com/spiffe/helm-charts/actions/runs/4808535068

for the result at the summary page.

The checks fail because I didn't update chart docs as intended, that commit bumping the chart version has been removed from this branch once, I saw it nicely executed the workflow.

Please note to merge #242 first. This PR will squash on top of that and therefore doesn't need another rebase, merge.

### CODE REVIEW

File: .github/tests/upstream-authority-cert-manager/post-install.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+EOF
+
+cat <<EOF >>"$GITHUB_STEP_SUMMARY"
+### spire
+
+| workload | Status |
+| -------- | ------ |
+| spire-server | $("${k_rollout_status[@]}" "${NS}" statefulset spire-server) |
+| spire-spiffe-csi-driver | $("${k_rollout_status[@]}"  "${NS}" daemonset spire-spiffe-csi-driver) |
+| spire-agent | $("${k_rollout_status[@]}" "${NS}" daemonset spire-agent) |
```
Comment (faisal-memon) [inline .github/tests/upstream-authority-cert-manager/post-install.sh:27]: oidc?

File: .github/tests/upstream-authority-cert-manager/post-install.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+k_rollout_status=(kubectl rollout status --watch --timeout 30s --namespace)
+
+cat <<EOF >>"$GITHUB_STEP_SUMMARY"
+### cert-manager
+
+| workload | Status |
+| -------- | ------ |
+| cert-manager | $("${k_wait[@]}" cert-manager deployments.apps cert-manager) |
+| cert-manager-cainjector | $("${k_wait[@]}" cert-manager deployments.apps cert-manager-cainjector) |
+| cert-manager-webhook | $("${k_wait[@]}" cert-manager deployments.apps cert-manager-webhook) |
```
Comment (faisal-memon) [inline .github/tests/upstream-authority-cert-manager/post-install.sh:17]: Not following this, we aren't deploying cert-manager.

File: .github/tests/upstream-authority-cert-manager/post-install.sh (hunk 1 above)
Comment (marcofranssen) [reply .github/tests/upstream-authority-cert-manager/post-install.sh:27]: This test doesn't have oidc enabled.

**Comment by faisal-memon** [comment]
The post-install.sh scripts are all duplicated. Can we have a common snippet somewhere thats imported?

**Comment by kfox1111** [comment]
This is looking good. Should we move this stuff to common.sh too? https://github.com/spiffe/helm-charts/blob/main/.github/tests/namespace-override/post-install.sh#L8-L14

File: .github/tests/federation-bundle-endpoint/post-install.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+set -x
+
+SCRIPT=$(readlink -f "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+NS="${SCRIPTPATH##*/}"
```
Comment (kfox1111) [inline .github/tests/federation-bundle-endpoint/post-install.sh:7]: NS is the same thing as $scenario. Should probably just use that, or add NS up at the workflow level.

File: .github/tests/common.sh [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  cat <<EOF >>"$GITHUB_STEP_SUMMARY"
+### $1
+
+| workload                             | Status |
+| ------------------------------------ | ------ |
+| spire-server                         | <pre>$(k_rollout_status "$1" statefulset spire-server)</pre> |
+| spire-controller-manager             | <pre>$(k_rollout_status "$1" statefulset spire-controller-manager)</pre> |
+| spire-spiffe-csi-driver              | <pre>$(k_rollout_status "$1" daemonset spire-spiffe-csi-driver)</pre> |
+| spire-agent                          | <pre>$(k_rollout_status "$1" daemonset spire-agent)</pre> |
+| spire-spiffe-oidc-discovery-provider | <pre>$(k_rollout_status "$1" deployments.apps spire-spiffe-oidc-discovery-provider)</pre> |
```
Comment (kfox1111) [inline .github/tests/common.sh:46]: These all use $1, assuming there is only ever one namespace. Might be more reusable to define $1 and $2 so split namespaces work too?

File: .github/tests/production-example/post-install.sh [hunk 5]
```diff
@@ -5,47 +5,24 @@ set -x
 SCRIPT=$(readlink -f "$0")
 SCRIPTPATH=$(dirname "$SCRIPT")
 
-k_wait=(kubectl wait --for condition=available --timeout 30s --namespace)
-k_rollout_status=(kubectl rollout status --watch --timeout 30s --namespace)
+# shellcheck source=/dev/null
```
Comment (kfox1111) [inline .github/tests/production-example/post-install.sh:8]: What does this do?

Comment (marcofranssen) [reply .github/tests/production-example/post-install.sh:9]: https://www.shellcheck.net/wiki/SC1091

**Comment by marcofranssen** [comment]
This PR got rebased on top of the shellcheck one #242 so I could eliminate some common mistakes.

**Review by kfox1111 — APPROVED** [review]
LGTM

**Pull request merged**